### ISSUE TITLE
Increased memory usage after migrating to tsi

### ISSUE THREAD

**Issue by hpbieker** [influxdata/influxdb#9653]
### Bug report

__System info:__ 
Ubuntu Linux 16.04
InfluxDB  1.5.1
128 GB RAM / 16 cores

__Steps to reproduce:__

Stop influxd
Run influx_inspect  buildtsi
Start influxd

__Expected behavior:__ 
Heap memory consumption should be less than without tsi.

__Actual behavior:__ 

Heap memory consumption increased from 30 GB to 55 GB after using tsi.

See graph below. I turned off influx at 20:00, ran buildtsi, and started influx.

![image](https://user-images.githubusercontent.com/24992762/38073908-428b1be0-332d-11e8-8604-405cf87bd9c6.png)

__Additional info:__ 
I also upgraded from influx 1.5.0 to influx 1.5.1.

[profiles.tar.gz](https://github.com/influxdata/influxdb/files/1858816/profiles.tar.gz)

There are 33256 L0-00000001.tsl files (no other L0-0000XXXX.tsl) and the same number of MANIFEST files.

**Comment by hpbieker** [comment]
This might be the same problem as reported in #9540, but in that case it was an upgrade from Influx 1.4 and on CentOS.

**Comment by hpbieker** [comment]
Upgrading to 1.5.2 improves things a bit, but now too much (1.5.2 is running after 19:45, 1.5.1 before that):
![image](https://user-images.githubusercontent.com/24992762/38753716-95eaa7d6-3f5f-11e8-8551-e772b34e2d04.png)

**Comment by nkipin-IX** [comment]
We are experiencing a similar issue with a newer version of influxdb.

Influxdb release: v1.6.3
OS: CentOS Linux release 7.5.1804 (Core)

Before "upgrading" to `tsi1` we saw ~6GB of memory used by influxdb.
After running `influx_inspect buildtsi`, changing the config to `tsi1` and restarting influxdb we saw memory usage by influxdb jump to ~18GB.

In an effort to resolve this we upgraded influxdb to v1.7.3 which saw another (slight) increase in memory usage.

![influxmem_rev](https://user-images.githubusercontent.com/47535287/52581443-c1519780-2df8-11e9-82c9-5c74804e82dd.png)
The red area is the time we took to run `influx_inspect buildtsi` and restart influxdb. The pink area is when we upgraded to v1.7.3
